### ISSUE TITLE
"Strange Numbers" in Galaxy Report - "Jobs per User"

### ISSUE THREAD

**Issue by quacksawbones** [galaxyproject/galaxy#13117]
When I enable and access the Galaxy Reports (via <galaxy_url>:9001), I see three numbers at the top of the page (which I presume are fetch times from the database or something similar).

Galaxy Version: 22.01 (dev0)
Commit: e25e2fa81ca02d929cedef09d97c35fbf57100fd

1. Install prerequisites (Python 3.9.9, PostgresQL 14.1)
    - NB: I have not tested this when the underlying database is MySQL or SQLite3.
2. Configure the firewall and open ports 8080 and 9001 to non-local connections
2. Download Galaxy (22.01, or dev at the time of reporting)
3. Configure galaxy.yml and report.yml
    - NB: The only things changed were:
        - `database_connection` (to connect to the Galaxy database using postgres)
        - `http` (to allow non-local web connections on ports 8080 and 9001 respectively)
4. Run Galaxy and Galaxy Reports
5. Navigate to <galaxy_url>:9001
6. In the list of reports, select "Jobs per User" under the "Jobs" section
7. Numbers should be displayed at the top of the main page content (above the report title)

NB: I have not tested this when the database is MySQL or SQLite3.

**Comment by quacksawbones** [comment]
I'm happy to help with this one (or at least have a dig around), but I will wait until after release_22.01.

**Comment by gregvonkuster** [comment]
I believe these are a remnant of debugging, although I'm not absolutely sure.  In any case, I'll soon have a PR that removes them - I'm close to finishing an enhancement to the reports, and I've removed the line in the template that displays these.

**Comment by bgruening** [comment]
What features do you need from this app? We do not maintain it well, and I guess we would be happy to drop it from the codebase. The bigger Galaxy servers are using the TIG Stack (Telegraf, InfluxDB, and Grafana) to display metrics and monitoring information. Do you think you could use something similar to address your needs?

**Comment by gregvonkuster** [comment]
>  I guess we would be happy to drop it from the codebase. 
I just submitted a PR with an enhancement to the reports, so I hope it can remain in the code base.  The PR fixes this issue as well.  :)

**Comment by bgruening** [comment]
I have seen it, but it still uses technologies from 10 years ago and no one cares to port it to Vue or other things which we are using in the other codebase.

**Comment by gregvonkuster** [comment]
I wasn't aware that the developers have moved on from the reports.  If the plan is to eliminate the reports, then please do it so that time is not wasted on enhancements and fixes.  

I'm running an instance where users need to be billed for jobs running on nodes that have various costs, so the PR I submitted enables capturing metrics for that on a monthly basis.    From your comments, I found this https://training.galaxyproject.org/training-material/topics/admin/tutorials/monitoring/tutorial.html, but these apps seem to be focused on monitoring.  Is it also possible to capture metrics for monthly job execution times per user per node type?

**Comment by bgruening** [comment]
We have __not__ deprecated it. I have just asked which features you are using and if this could be replaced by the TIG stack. 

Have a look at https://stats.galaxyproject.eu/ and play around with that side.

If you change
![grafik](https://user-images.githubusercontent.com/469983/148555649-089fd7e9-a2c9-4fc2-a695-97f41e410534.png)
You can see many more dashboards.

**Comment by hexylena** [comment]
> but these apps seem to be focused on monitoring. Is it also possible to capture metrics for monthly job execution times per user per node type?

yeah those queries are built into [`gxadmin`](https://github.com/galaxyproject/gxadmin) (or could be) and be a bit more scriptable for your billing processes. 

![image](https://user-images.githubusercontent.com/458683/148741691-d10bb8fc-5b23-4e07-a602-cd1e658beeef.png)

I don't see anything that maps directly to your query but it's 90% of the way there with existing queries and it's very easy to add your own query.

**Comment by gregvonkuster** [comment]
Thanks @hexylena - I haven't had a chance to look at these other apps, but I'll get to them.  What is the plan for the reports app?  Is it going to be deprecated?  If so, will it remain in the code base or be removed from it.  Beyond the move to "legacy_framework" (https://github.com/galaxyproject/galaxy/tree/dev/lib/galaxy/web/legacy_framework), I didn't see any comments in the code that it should not be enhanced or maintained or that the plan was to eliminate it and that it has been replaced by one or more of these external apps.  I easily could have missed something though - I've not been involved in Galaxy framework code for quite some some time now...

**Comment by gregvonkuster** [comment]
> yeah those queries are built into [`gxadmin`](https://github.com/galaxyproject/gxadmin) (or could be) and be a bit more scriptable for your billing processes.

I've taken a look at gxadmin and I can see that enhancements can certainly be made so that it can handle the queries I need.  Of course, this requires the same work which was just added to the Galaxy reports.  :)

One of the very nice things about the Galaxy reports is that the users have direct access to them, and so have the ability to directly confirm monthly charges, etc.  I'm sure that when some users get a bill, they'll be shocked to learn that their analyses took that many node hours, or that their lab ran that many jobs over the past month.  And then in some cases, they'll want some kind of proof, and gxadmin is not available to them.

I've so far only briefly looked at the other 3 web apps for monitoring that have replaced the reports (Telegraf, InfluxDB, and Grafana), but from what I've seen, most users will get lost trying to use them to provide the same information as the more simple-to-use Galaxy reports.

I can understand why the Galaxy reports are not being maintained by the developers.  But I feel that they are valuable enough that fixes and enhancements should be accepted for them, especially since they remain a part of the code base.

**Comment by bgruening** [comment]
@gregvonkuster no one has said that the PR will not be merged. I was only interested to know about usecases and who is using this app. If there are use cases that we can not replace, we should keep the app and you can help maintaining it :)

The billing use-case is an interesting one and a few people have talked about it. E.g. different locations can have a different coast and the job scheduling should take this into account. Independent of this issue, maybe you can describe your use-case in a GDoc or as an issue.

**Comment by gregvonkuster** [comment]
Thanks @bgruening!

The use-case for my current PR is fairly straight-forward, just billing per user per node type (cost can be different per node type) per month.  I initially thought I'd have to configure Galaxy to run jobs as the logged in user, but @natefoo told me that is complex, not really supported, and not necessary for this kind of billing.

I have another use case for the Galaxy reports web app though, and this one is critical for the Galaxy / CoralSNP instance at https://coralsnp.science.psu.edu/galaxy/.  The "regular" reports for this instance are available here https://coralsnp.science.psu.edu/jobs/.  

But this Galaxy instance also includes a separate Postgres database that we've named stag, and I've leveraged the framework reports web app to create another "reports" web app that is available here https://coralsnp.science.psu.edu/reports.  This set of reports provides query results from this stag database.  These reports are critical for the Galaxy / CoralSNP instance, which would become useless without them.  And Galaxy / CoralSNP users need to have public access to these stag database reports.

I'm happy to document this more fully in a location that is more meaningful.  Just let me know where you feel it should be.
